### PR TITLE
add: debian package build step

### PR DESCRIPTION
**Pull request by neo773**
This PR adds build step for Debian package.

Tested on Debian 12
SCR-20240106-ehgf

/claim #3335

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: .github/workflows/release-artifacts.yml [hunk 1]
```diff
@@ -56,6 +56,15 @@ jobs:
...
           target: ${{ matrix.target }}
           token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build Debian package
+        run: |
+          cargo install cargo-deb
+          cargo deb --no-build
+      - name: Upload Debian package
+        uses: actions/upload-artifact@v2
+        with:
+          name:  ${{ matrix.target }}-debian
```
Comment (generall) [inline .github/workflows/release-artifacts.yml:66]: It looks like you are building it in `build-mac-binaries` for mac os. It should be in `build-linux-binaries`

**Review by generall — CHANGES_REQUESTED** [review]
Could you please trigger CI on your fork, so we could see that CI actually works and produces usable packages

**Comment by neo773** [comment]
@generall 

Successful build [here](https://github.com/neo773/qdrant/actions/runs/7428502193/job/20215913616)

SCR-20240106-gxyh

**Review by generall — CHANGES_REQUESTED** [review]
When I try to install it on my server, i see this error:
image

As I understood, deb can't automatically install all required dependencies.

Would it be possible to base .deb package on MUSL build from here https://github.com/qdrant/qdrant/pull/3342 to avoid errors like this?

**Comment by neo773** [comment]
Looking into MUSL, worked fine on my Ubuntu VM so I assumed it was good.

image

**Comment by neo773** [comment]
Pushed a new commit that uses musl, also tested the package on x86 this time.

image


[Release](https://github.com/neo773/qdrant/releases/tag/v1.7.10)
[CI Run](https://github.com/neo773/qdrant/actions/runs/7434325055/job/20228354047#step:7:146)

**Comment by neo773** [comment]
Sorry, I just realized I was merging into `master` instead of `dev` closing this PR as I've made a new one https://github.com/qdrant/qdrant/pull/3344

**Pull request closed without merging**